gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -246,6 +247,10 @@ public Boolean load(String bucketName) {
               .setDaemon(true)
               .build());
 
+  // Thread-pool used for background watchdog
+  private final ScheduledExecutorService scheduledExecutorService =
+      newSingleThreadScheduledExecutor();
```

Comment:
We already have `backgroundTasksThreadPool` that we use for different background tasks, can we reuse it?

Reply:
We need an instance of `ScheduledExecutorService` for this usecase